src/components/navigation/TableOfContents.tsx

```
@@ -0,0 +1,76 @@
+          .filter(e => e.isIntersecting)
+          .sort((a, b) => a.boundingClientRect.top - b.boundingClientRect.top)[0];
+        if (visible?.target?.id) {
+          setActiveId(visible.target.id);
+        }
+      },
+      { rootMargin: '-20% 0px -70% 0px', threshold: [0, 1] }
+    );
+
+    // Observe all heading elements
+    const headings = document.querySelectorAll('h1, h2, h3, h4, h5, h6');
+    headings.forEach(el => observerRef.current?.observe(el));
```

Biome (2.3.14)

[error] This callback passed to forEach() iterable method should not return a value.

Either remove this return or remove the returned value.


(lint/suspicious/useIterableCallbackReturn)